nuget/chocolatey/tools/chocolateysetup.psm1

```diff
@@ -30,13 +30,54 @@ param (
   }
 }
 
+function Remove-ShimWithAuthenticodeSignature {
+  param (
+    [string] $filePath
+  )
+  if (!(Test-Path $filePath)) {
+    return
+  }
+
+    $signature = Get-AuthenticodeSignature $filePath -ErrorAction SilentlyContinue
+
+    if (!$signature -or !$signature.SignerCertificate) {
+      Write-ChocolateyWarning "Shim found in $filePath, but was not signed. Ignoring removal..."
+      return
+    }
+
+    $possibleSignatures = @(
+      'RealDimensions Software, LLC'
+      'Chocolatey Software, Inc\.'
+    )
+
+    $possibleSignatures | ForEach-Object {
+      if ($signature.SignerCertificate.Subject -match "$_") {
+        Write-Output "Removing shim $filePath"
+        $null = Remove-Item "$filePath"
+
+        if (Test-Path "$filePath.ignore") {
+          $null = Remove-Item "$filePath.ignore"
+        }
+
+        if (Test-Path "$filePath.old") {
+          $null = Remove-Item "$filePath.old"
+        }
+      }
+    }
+
+    if ($signature.SignerCertificate.Thumbprint -ne '83AC7D88C66CB8680BCE802E0F0F5C179722764B') {
+      Write-ChocolateyWarning "Shim found in $filePath, but did not match our signature. Ignoring removal..."
+      return
+    }

```

Comment:
Is this part needed if we're removing the file on line 56?

Reply:
Good catch, seems I didn't remove this one after refactoring the method to support multiple Subjects.
I'll get that part removed.